### PR TITLE
XW-1208 | Fix Quantserve late loading

### PR DESCRIPTION
**Pull request by vforge**
## Links

* http://wikia-inc.atlassian.net/browse/XW-1208

## Description

We're moving some initial logic for Quantserve PV to the server so we'll no longer wait for Ember.

## Reviewers

@Wikia/x-wing

### CODE REVIEW

File: server/app/views/_partials/dns-prefetch.hbs [hunk 1]
```diff
@@ -3,6 +3,7 @@
 {{/if}}
 {{#unless queryParams.noexternals}}
 	<link rel="dns-prefetch" href="{{server.gaUrl}}">
+  <link rel="dns-prefetch" href="http://edge.quantserve.com">
```
Comment (bognix) [inline server/app/views/_partials/dns-prefetch.hbs]: use tabs not spacing

Comment (rogatty) [inline server/app/views/_partials/dns-prefetch.hbs]: `\t`

File: front/common/modules/Trackers/Quantserve.js [hunk 2]
```diff
@@ -1,45 +0,0 @@
...
-	url() {
-		const prefix = (document.location.protocol === 'https:' ? 'https://secure' : 'http://edge');
-
-		return `${prefix}.quantserve.com/quant.js?${Math.random()}`;
-	}
-
-	/**
-	 * @returns {void}
-	 */
-	trackPageView() {
```
Comment (rogatty) [inline front/common/modules/Trackers/Quantserve.js]: Won't we stop tracking subsequent page views?

**Comment by bognix** [comment]
add support for tracking pageviews in Ember and what about unit tests?

**Comment by bognix** [comment]
LGTM

File: front/common/modules/Trackers/Quantserve.js (hunk 2 above)
Comment (vforge) [inline front/common/modules/Trackers/Quantserve.js]: Nope, I've added that in https://github.com/Wikia/mercury/commit/6cf99108cd7baa489154540e16a61bc40bdb670c

Comment (rogatty) [inline front/common/modules/Trackers/Quantserve.js]: Shouldn't we leave the tracker module and put `window.trackQuantservePageView();` here in `Quantserve#trackPageView()`?

After this change Quantcast would be an exception. We'll have to modify other trackers soon, so maybe we should plan ahead and have some standarized way to track page views? Also, this PR deletes a unit test and doesn't replace it with something else. Not good :-1:

File: server/app/views/_partials/server-data.hbs [hunk 3]
```diff
@@ -6,6 +7,7 @@
 	M.prop('translations', {{{json (translations wikiVariables.language.content ns=localSettings.translationFiles)}}});
 	M.prop('weppyConfig', {{{json localSettings.weppy}}});
 	M.prop('wikiaLanguage', '{{{wikiVariables.language.content}}}');
+	M.prop('wikiaVertical', '{{{wikiVariables.vertical}}}');
```
Comment (rogatty) [inline server/app/views/_partials/server-data.hbs]: Why not using `Mercury.wiki.vertical`? What's the duplicate for?

File: server/app/views/_partials/tracking-quantserve.hbs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+			// without this quantserve does not want to track 2+ page view
+			window.__qc = null;
+
+			window._qevents = [{
+				qacct: M.prop('tracking.quantserve'),
+				labels: quantcastLabels.join(',')
+			}];
+
+			scriptTag.async = 1;
+			scriptTag.src = prefix + '.quantserve.com/quant.js?' + Math.random();
```
Comment (hakubo) [inline server/app/views/_partials/tracking-quantserve.hbs]: can we load this script once?

File: server/app/views/_partials/tracking-quantserve.hbs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+				firstScriptInDocument = document.getElementsByTagName('script')[0];
+
+			if (M.prop('wikiaVertical')) {
+				quantcastLabels.unshift(M.prop('wikiaVertical'));
+			}
+
+			// without this quantserve does not want to track 2+ page view
+			window.__qc = null;
+
+			window._qevents = [{
```
Comment (hakubo) [inline server/app/views/_partials/tracking-quantserve.hbs]: I think if you used push here, 
you won't have to dowload scripts X times

File: front/common/modules/Trackers/Quantserve.js (hunk 2 above)
Comment (vforge) [inline front/common/modules/Trackers/Quantserve.js]: Talked offline

File: server/app/views/_partials/tracking-quantserve.hbs (hunk 4 above)
Comment (vforge) [inline server/app/views/_partials/tracking-quantserve.hbs]: After a small test - no.

File: server/app/views/_partials/tracking-quantserve.hbs (hunk 5 above)
Comment (vforge) [inline server/app/views/_partials/tracking-quantserve.hbs]: See comment above.

**Comment by tomasznapieralski** [comment]
LGTM

**Comment by hakubo** [comment]
LGTM

**Comment by tomasznapieralski** [comment]
Tested

**Pull request merged**